docs/proposals/cluster-autoscaling.md

```diff
@@ -0,0 +1,314 @@
+start again leaving the cluster in the fixed, too small size.
+
+## Kuberntes 1.2 - small KAC
+
+For Kubernetes 1.2 we plan to modify the architecture slightly. The main difference is moving the config to the Kubernetes world.
+The config will be placed in apiserver (or in the other config storage that will be ready for 1.2)  and will be handled by
+kubernetes-specific tools (kubectl or so). A new kubernetes controller will be added - Kubernetes Autoscaler for Cluster (abbreviated as KAC).
+
+In version 1.2 we will provide “small” KAC. Its responsibility will be mainly limited to reading the config and updating Google
+Cloud Autoscaler accordingly. It will also observe changes in MIG and generate appropriate kubernetes events.
+
+```
+                          +--MIG-----------------------------+
+                          |                                  |
+                          | +--------+ +--------+ +--------+ |
+                          | |        | |        | |        | |
+                          | | Node1  | | Node2  | | NodeX  | |
+                          | |        | |        | |        | +<--+
+                          | +--------+ +--------+ +--------+ |   |
+            +---------+   | |cAdvisor| |cAdvisor| |cAdvisor| |   |
+            |  Config |   | +--------+ +--------+ +----+---+ |   |
+            +---------+   |     |          |           |     |   |
+                 |        +-----|----------|-----------|-----+   |
+                 |              |          |           |         |
+            +----v----+         |      +---v----+      |         |
+            |   KAC   |         +----> |Heapster| <----+         |
+            +---------+                +---+----+                |
+                 |                         |                     |
+ Kubernetes      |                         |                     |
+-----------------|-------------------------|---------------------|------
+  Google         |                         |                     |
+                 |                         |                     |
+                 |                  +------v-----+       +-------+------+
+                 |                  |Google Cloud|       |MIG Definition|
+                 |                  | Monitoring |       +-------^------+
+                 |                  +------------+               |
+                 |                         |                     |
+                 |                         |                     |
+                 |                  +------v-----+               |
+                 +----------------->|Google Cloud+---------------+
+                                    | Autoscaler |
+                                    +------------+
+```
+
+
+We will also configure GCE Autoscaler so that it deletes nodes  in a non-random way. There are couple strategies for killing to be considered:
+
```

Comment:
Since this is hypothetical, please add the one I had suggested, "node with fewest number of guaranteed-QoS pods"